docs/react/react-native.md

```diff
@@ -76,3 +76,52 @@ export function useRefreshOnFocus<T>(refetch: () => Promise<T>) {
 ```
 
 In the above code, `refetch` is skipped the first time because `useFocusEffect` calls our callback on mount in addition to screen focus.
+
+## Disable re-renders on out of focus Screens
+
+In some situations, including performance concerns, you may want to stop re-renders when a React Native screen gets out of focus. To achieve this we can use `useFocusEffect` from `@react-navigation/native` together with the `notifyOnChangeProps` query option.
+
+This custom hook provides a `notifyOnChangeProps` option that will return an empty array whenever a screen goes out of focus - effectively stopping any re-renders on that scenario. Whenever the screens gets in focus again, the behavior goes back to normal.
+
+```tsx
+import React from 'react'
+import { UseQueryOptions } from '@tanstack/react-query'
+import { useFocusEffect } from '@react-navigation/native'
+
+export function useFocusNotifyOnChangeProps(
+  notifyOnChangeProps: UseQueryOptions['notifyOnChangeProps']
+) {
+  const focusedRef = React.useRef(true)
+  const focusedNotifyOnChangePropsRef = React.useRef(notifyOnChangeProps)
+
+  React.useLayoutEffect(() => {
+    focusedNotifyOnChangePropsRef.current = notifyOnChangeProps
+  })
+
+  useFocusEffect(
+    React.useCallback(() => {
+      focusedRef.current = true
+
+      return () => {
+        focusedRef.current = false
+      }
+    }, [])
+  )
+
+  return React.useCallback(() => {
+    if (!focusedRef.current) {
+      return []
+    }
+
+    if (typeof focusedNotifyOnChangePropsRef.current === 'function') {
+      return focusedNotifyOnChangePropsRef.current()
+    }
+
+    return focusedNotifyOnChangePropsRef.current
+  }, [])
+}
+```
```

Comment:
Would be good to give an example on how this would actually be used:

```tsx
const MyExampleComponent = () => {
  const notifyOnChangeProps = useFocusNotifyOnChangeProps('all');
  const { dataUpdatedAt } = useQuery({
    queryKey: ['myExampleKey'],
    queryFn: async () => {
      const response = await fetch('https://api.github.com/repos/tannerlinsley/react-query');
      return response.json();
    },
    notifyOnChangeProps,
  });

  return <div>DataUpdatedAt: {dataUpdatedAt}</div>;
};
```

Reply:
```
const notifyOnChangeProps = useFocusNotifyOnChangeProps('all');
```

`'all'` is actually not the default, but `undefined` is, which does smart tracking of used properties.

Reply:
Changed to `undefined` 👍